packages/edit-site/src/components/list/actions/rename-menu-item.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import { useEntityProp, store as coreStore } from '@wordpress/core-data';
+import { __ } from '@wordpress/i18n';
+import {
+	Button,
+	Flex,
+	FlexItem,
+	MenuItem,
+	Modal,
+	TextControl,
+} from '@wordpress/components';
+
+export default function RenameMenuItem( { template, onClose } ) {
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+
+	const { saveEditedEntityRecord } = useDispatch( coreStore );
+	const [ title, setTitle ] = useEntityProp(
+		'postType',
+		template.type,
+		'title',
+		template.id
+	);
+
+	async function onTemplateRename( event ) {
+		event.preventDefault();
+		setIsModalOpen( false );
+		onClose();
+
+		// Persist edited entity.
+		await saveEditedEntityRecord( 'postType', template.type, template.id );
```

Comment:
We should probably also add a `isLoading` state and set `isBusy` to the submit button. And once the change is saved, we can then close the modal and potentially display success or error notices as in https://github.com/WordPress/gutenberg/pull/36808.

Reply:
I don't think there's a need for the busy state. I'm closing the modal before the HTTP request is made to persist changes. So there shouldn't be a visible busy state.

I've updated the code to display notices.

P.S. There should be an easier way to use `try/catch` with actions.